sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -22,14 +22,67 @@
 
 from __future__ import absolute_import
 
-from apache_beam import coders
+from apache_beam.io.iobase import Read
+from apache_beam.io.iobase import Write
 from apache_beam.runners.dataflow.native_io import iobase as dataflow_io
+from apache_beam.transforms import PTransform
+from apache_beam.transforms import ParDo
 from apache_beam.transforms.display import DisplayDataItem
 
-__all__ = ['PubSubSink', 'PubSubSource']
 
+__all__ = ['ReadStringsFromPubSub', 'WriteStringsToPubSub']
 
-class PubSubSource(dataflow_io.NativeSource):
+
+class ReadStringsFromPubSub(PTransform):
```

Comment:
I would really prefer for `ReadFrom*` to be consistent across Python SDK primitives and for this class name to not have the `String` substring so that this can be less verbose, but I do agree that we could potentially also have `ReadMessagesFromPubSub` in the future which returns `PubSubMessage` objects for more context (however, in that case, I would argue we should only have the `ReadMessagesFromPubSub` form so that the user could inspect the resulting `PubSubMessage` message and do whatever they want with that production).

Even in this case, though, could you add the coder argument since there are different types of string encodings in Python?  For example, the PubSub string messages may be encoded in some other encoding like ascii or utf-16, for which we would like to have an API similar to that for ReadFromText, rather one that tries to enumerate Read<TypeOfString>StringFromText for all possible encodings.

CC: @robertwb -- thoughts?